plugins/node/opentelemetry-instrumentation-pg/src/instrumentation.ts

```diff
@@ -297,10 +297,11 @@ export class PgInstrumentation extends InstrumentationBase<PgInstrumentationConf
 
         // Modify query text w/ a tracing comment before invoking original for
         // tracing, but only if args[0] has one of our expected shapes.
+        // Also omit the tracing comment if args[0] is a prepared query object.
         if (instrumentationConfig.addSqlCommenterCommentToQueries) {
           args[0] = firstArgIsString
             ? addSqlCommenterComment(span, arg0)
-            : firstArgIsQueryObjectWithText
+            : firstArgIsQueryObjectWithText && !('name' in arg0)
```

Comment:
This section is already quite complex IMO with the double js ternary operator.

While the comment above is useful, I can see how future readers might get confused figuring out that the `name` in `arg0` when it's an object is an indication for prepared statement.

The link you gave in the PR body says that:

> PostgreSQL has the concept of a [prepared statement](https://www.postgresql.org/docs/9.3/static/sql-prepare.html). node-postgres supports this by supplying a name parameter to the query config object

I suggest refactoring this part to make it clear why this check is applied

Reply:
Should be a lot clearer now.